### ISSUE TITLE
Homework 16-17. AJAX

### ISSUE THREAD

**Issue by mrbalov** [Plyavinskiy/FE_online_2020_pliavinski#1]
## Questions/Remarks

- **Using of `index.js` is incorrect.**

In your `/js/index.js` file a lot of inlined logic. This is wrong. `index.js` file should be used for several purposes:
- entity(ies) re-exporting;
- initializations stuff (using imported initializer(s)).

Also your code in `index.js` file looks to imperative. Read below about another issues.

- **Separate server communications from representational code.**
You use server communications (`fetch` function) right with a representational code (i.e. HTML). It is wrong. Server communications should be separated into independent module. The simplest solution: you can create `API` class with all the server communication methods you need. Please, explore [MVC and similar patterns](https://habr.com/ru/post/215605/). The idea is to divide code into several layers responsible for a different things.

E.g., following code

```js
data.header.forEach((item) => {
  headerLogo.innerHTML = `<a class="header__logo" href="#">${item.logo}</a>`;
  subtitleTitleHeader.innerHTML = `
    <h1 class="hero__title">
      ${item.title}
    </h1>
    <p class="hero__description">
      ${item.description}
    </p>
    <button class="hero__button">${item.btn1}</button>
    <button class="hero__button btn-inverse">${item.btn2}</button>
  `;
});
```
might be represented as function:

```js
function renderHeaderItems(items) {
  items.forEach((item) => {
    headerLogo.innerHTML = `<a class="header__logo" href="#">${item.logo}</a>`;
    subtitleTitleHeader.innerHTML = `
      <h1 class="hero__title">
        ${item.title}
      </h1>
      <p class="hero__description">
        ${item.description}
      </p>
      <button class="hero__button">${item.btn1}</button>
      <button class="hero__button btn-inverse">${item.btn2}</button>
    `;
  });
}
```

- **Code is too imperative.**
You have a lot of mixed logic in the `/js/index.js` file. You should divide all this code into small functions with explicit names. Such atomic code is more understandable, app gets more agile and extensible.

- **Code style improvements.**

There are some problems with code style. Firstly, don't use "early return" style. The following code

```js
function validateTitle(title) {
  if (typeof title !== 'string') {
    return 'Incorrect input data';
  }

  if (title.length < 2 || title.length > 20) {
    return 'INVALID';
  }

  // ...

  return isValidChars ? 'VALID' : 'INVALID';
}
```

should be rewritten into

```js
function validateTitle(title) {
  if (typeof title !== 'string') {
    return 'Incorrect input data';
  } else if (title.length < 2 || title.length > 20) {
    return 'INVALID';
  } else {
    // ...

    return isValidChars ? 'VALID' : 'INVALID';
  }
}
```

This code structure is easy to understand and extend.

To sum up, you should work on code style. You can follow, for example, [**Google JavaScript Style Guide**](https://google.github.io/styleguide/jsguide.html).

## Summary
Well done! These are main moments I want you to work on improvements.
